r-package/grf/R/average_treatment_effect.R

```diff
@@ -125,16 +125,13 @@ average_treatment_effect = function(forest,
   
   # Retreive pointwise treatment effect predictions from forest, and
   # compute naive average effect estimates (notice that this uses OOB)
-  forest.with.preds <- forest
-  
-  if (is.null(forest.with.preds$predictions)) {
-    forest.with.preds$predictions <- predict(forest.with.preds)$predictions
-    eval.parent(substitute(forest <- forest.with.preds))
-  }  
+  predictions <- predict(forest)$predictions
```

Comment:
Is it possible to allow the user to do this outside of the function and give the function these predictions? I'm thinking about a use case where one is trying to look at CATEs across multiple subgroups of multiple variables. I have a wrapper function that will take a forest and some variables and then look at CATEs for every level of each of those variables. This means that one has to generate predictions—the same predictions—every time. Allowing the user to specify this would allow the wrapper function to do it just once. Perhaps it's too focused on my specific use case, and I can edit the code myself.

Also, what is the significance that one is using out-of-bag and how do we know from the `predict` function that we are using out-of-bag?

For reference, here's that wrapper function:

```
# get conditional average treatment effects
  cates <- data.frame(var = rep(NA, 100), level = NA, 
                      cate = NA, lb = NA, ub = NA)
  counter <- 1
  for (i in seq_along(vars)) {
    for (j in levels(data[[vars[i]]])) {
      tmp <- cate(
        forest, 
        subset = data[, vars[i]] == j
      )
      cates$var[[counter]] <- vars[i]
      cates$level[[counter]] <- j
      cates$cate[[counter]] <- tmp[[1]]
      cates$lb[[counter]] <- tmp[[1]] - 1.96 * tmp[[2]]
      cates$ub[[counter]] <- tmp[[1]] + 1.96 * tmp[[2]]
      counter <- counter + 1
    }
  }
```